src/platform/Linux/DnssdImpl.cpp

```diff
@@ -383,13 +383,19 @@ void MdnsAvahi::HandleClientState(AvahiClient * client, AvahiClientState state)
         break;
     case AVAHI_CLIENT_FAILURE:
         ChipLogError(DeviceLayer, "Avahi client failure");
-        mErrorCallback(mAsyncReturnContext, CHIP_ERROR_INTERNAL);
+        // error happend, we should to restart dnssd.
+        mErrorCallback(mAsyncReturnContext, CHIP_ERROR_FORCED_RESET);
```

Comment:
If there's a real failure, won't this continuously restart avahi?

Reply:
Yes, if an error occurs, it will continue to restart. This is the same as restarting the service when a collision error occurs. It will still restart continuously.